dask/array/core.py

```diff
@@ -3129,7 +3129,9 @@ def normalize_chunks(chunks, shape=None, limit=None, dtype=None, previous_chunks
                 "Got chunks=%s, shape=%s" % (chunks, shape)
             )
 
-    return tuple(tuple(int(x) if not math.isnan(x) else x for x in c) for c in chunks)
+    return tuple(
+        tuple(int(x) if not math.isnan(x) else np.nan for x in c) for c in chunks
+    )
```

Comment:
It looks like existing code that uses `float("nan")` in `rechunk(...)` calls will continue to work. Just wanted to double check that that's indeed the case.

Reply:
Yes, that's tested with the second test case I added. 

If someone had somehow set the existing chunks of an array to `float("nan")`, my code wouldn't catch that, but from what I can see, this _should_ not happen if you're using things correctly. In that case, the code should still behave as it did before.